### PR TITLE
Save model predictions for a variety of scenarios for later analysis.

### PR DESCRIPTION
**Pull request by zack112358**
In this change, just the collection of model outputs for a grid of various inputs. Comparison tk soon with a jupyter notebook or the like. Can you recommend an older revision to try this change against to see what effect past model changes have had?

Regarding the issue I had previously with running scripts: I wound up just going with a test that dumps out a file as a side effect. It was easier than getting too into it.

### CODE REVIEW

**Comment by apiology** [comment]
> Can you recommend an older revision to try this change against to see what effect past model changes have had?

Good question.  Based on the upper right updates section, it looks like some updates for Omicron went in Jan 4, 2022, so looking around that time range might be one idea.

There was also a PR merged and then reverted recently - see #1421 and #1693.

> Regarding the issue I had previously with running scripts: I wound up just going with a test that dumps out a file as a side effect. It was easier than getting too into it.

Makes sense to me!

File: src/data/__tests__/scenarios.test.ts [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+          row.sanitized = sanitizeData(row.data)
+          row.result = calculate(row.sanitized)
+          results.push(row)
+        }
+      }
+    }
+    let fd
+
+    try {
+      fd = fs.openSync('computed_scenarios.json', 'w')
```
Comment (apiology) [inline src/data/__tests__/scenarios.test.ts:367]: Can you add this file to .gitignore?

**Review by apiology — COMMENTED** [review]
This is great progress towards a powerful goal--thanks so much for digging in on it!  Thoughts:

1. If this is enough for any kind of rough and ready use, a comment or two or a line in README.md would be appreciated on how to use it or at least what it is and what it does.  If it's just incremental progress to share or checkpoint, no worries!

2. Let me know if you could use a hand on the Typescript errors.

3. Are the yarn.lock changes necessary?  I didn't see any corresponding package.json changes.  I wonder iif they might be breaking the netlify build:

https://app.netlify.com/sites/microcov/deploys/63cb654170771300088b86ef

```
11:09:48 PM: yarn run v1.22.4
11:09:48 PM: $ REACT_APP_DEPLOY_PRIME_URL="$DEPLOY_PRIME_URL" react-scripts build
11:09:49 PM: Creating an optimized production build...
11:09:50 PM: Failed to compile.
11:09:50 PM: 
11:09:50 PM: ./src/index.tsx
11:09:50 PM: Error: [BABEL] /opt/build/repo/src/index.tsx: Cannot find module '@babel/helper-builder-react-jsx'
11:09:50 PM: Require stack:
11:09:50 PM: - /opt/build/repo/node_modules/babel-preset-react-app/node_modules/@babel/preset-react/node_modules/@babel/plugin-transform-react-jsx/lib/transform-classic.js
11:09:50 PM: - /opt/build/repo/node_modules/babel-preset-react-app/node_modules/@babel/preset-react/node_modules/@babel/plugin-transform-react-jsx/lib/index.js
11:09:50 PM: - /opt/build/repo/node_modules/babel-preset-react-app/node_modules/@babel/preset-react/lib/index.js
11:09:50 PM: - /opt/build/repo/node_modules/babel-preset-react-app/create.js
11:09:50 PM: - /opt/build/repo/node_modules/babel-preset-react-app/index.js
11:09:50 PM: - /opt/build/repo/node_modules/react-scripts/node_modules/@babel/core/lib/config/files/plugins.js
11:09:50 PM: - /opt/build/repo/node_modules/react-scripts/node_modules/@babel/core/lib/config/files/index.js
11:09:50 PM: - /opt/build/repo/node_modules/react-scripts/node_modules/@babel/core/lib/index.js
11:09:50 PM: - /opt/build/repo/node_modules/react-scripts/node_modules/babel-loader/lib/index.js
11:09:50 PM: - /opt/build/repo/node_modules/loader-runner/lib/loadLoader.js
11:09:50 PM: - /opt/build/repo/node_modules/loader-runner/lib/LoaderRunner.js
11:09:50 PM: - /opt/build/repo/node_modules/webpack/lib/NormalModule.js
11:09:50 PM: - /opt/build/repo/node_modules/webpack/lib/NormalModuleFactory.js
11:09:50 PM: - /opt/build/repo/node_modules/webpack/lib/Compiler.js
11:09:50 PM: - /opt/build/repo/node_modules/webpack/lib/webpack.js
11:09:50 PM: - /opt/build/repo/node_modules/react-scripts/scripts/build.js (While processing: "/opt/build/repo/node_modules/babel-preset-react-app/index.js")
11:09:50 PM: error Command failed with exit code 1. (https://ntl.fyi/exit-code-1)
11:09:50 PM: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
11:09:50 PM: ​
11:09:50 PM:   "build.command" failed                                        
11:09:50 PM: ────────────────────────────────────────────────────────────────
11:09:50 PM: ​
11:09:50 PM:   Error message
11:09:50 PM:   Command failed with exit code 1: yarn build (https://ntl.fyi/exit-code-1)
11:09:50 PM: ​
11:09:50 PM:   Error location
11:09:50 PM:   In Build command from Netlify app:
11:09:50 PM:   yarn build
11:09:50 PM: ​
11:09:50 PM:   Resolved config
11:09:50 PM:   build:
11:09:50 PM:     command: yarn build
11:09:50 PM:     commandOrigin: ui
11:09:50 PM:     environment:
11:09:50 PM:       - REACT_APP_PUBLIC_URL
11:09:50 PM:       - REVIEW_ID
11:09:50 PM:     publish: /opt/build/repo/build
11:09:50 PM:     publishOrigin: ui
```

**Comment by apiology** [comment]
Thanks for fixing up CI!  Happy to give it a final look when you're ready.

**Comment by zack112358** [comment]
Assuming that CI passes, I think it's ready for re-review.

File: model_comparison.ipynb [hunk 2]
```diff
@@ -0,0 +1,197 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "03835d1a",
+   "metadata": {},
+   "source": [
+    "# To compare model outputs\n",
+    "This notebook is for comparing the computed_scenarios.json outputs from different model versions of MicroCovid. For example, you might want to compare the model output before and after a pull request is merged. To do that, run the tests before and after the change, save both versions of computed_scenarios.json under different names, and tell us what they are here (\"reference_file\" for before the change and \"test_file\" for after, is the intention).\n",
```
Comment (apiology) [inline model_comparison.ipynb:9]:
```suggestion
    "This notebook is for comparing the computed_scenarios.json outputs from different model versions of microCovid. For example, you might want to compare the model output before and after a pull request is merged. To do that, run the tests before and after the change, save both versions of computed_scenarios.json under different names, and tell us what they are here (\"reference_file\" for before the change and \"test_file\" for after, is the intention).\n",
```

**Review by apiology — APPROVED** [review]
Nice!  Thanks a ton for taking the time to put this together.

**Comment by zack112358** [comment]
For clarity: I am not authorized to merge to "main" so I'll need you to push the merge button.

**Comment by apiology** [comment]
Got it - must be more GitHub settings to fiddle with :)

**Pull request merged**